### ISSUE TITLE
determining the tag tilt deploys with

### ISSUE THREAD

**Issue by nicks** [tilt-dev/tilt#3584]
Suppose you have a custom_build.

Tilt will set the environment variable EXPECTED_REF. When the custom_build script finishes, Tilt looks in the docker image store to find the image. Then it retags the image with a content-based tag and deploys it.

For Tilt, it's important to deploy a content-based tag. The architecture assumes that the tag changes when the content of the image changes, and stays the same when the content doesn't change. KNative uses a similar strategy to make similar guarantees.

But there are cases where the custom_build script needs access to the content-based tag:
1) When another script in the build pipeline needs to use the same tag
2) When the container you're running needs to use the tag to deploy workers

We currently have a solution to (2) but it's not well-documented. You need to add an env var to your container:

```
env:
  - name: IMAGE_BUSYBOX
    value: busybox
```

and then in your Tiltfile

```
custom_build('busybox', ..., match_in_env_vars=True)
```

to tell Tilt to inject the image reference (with the right tag) into the env variable

We don't have a good solution to (1) right now, though you could imagine a simple command like

```
tilt dump image-deploy-ref MY_IMAGE
```

that the custom_build script can run after it finishes building the image, so that tilt can communicate back what it will tag the image with (you need to build the image first to determine what the tag will be)

**Comment by nicks** [comment]
(part of this might be resolved with a better airflow example -- the env variable is similar to how i'm running airflow pipelines with the KubernetesPodOperator)

**Comment by nicks** [comment]
this is released in tilt v0.17.0
